lib/browser/api/menu.js

```diff
@@ -45,30 +45,49 @@ Menu.prototype._init = function () {
 
 Menu.prototype.popup = function (window, x, y, positioningItem) {
   let [newX, newY, newPosition, newWindow] = [x, y, positioningItem, window]
+  let opts
 
   // menu.popup(x, y, positioningItem)
-  if (!window) {
-    // shift over values
-    if (typeof window !== 'object' || window.constructor !== BrowserWindow) {
-      [newPosition, newY, newX, newWindow] = [y, x, window, null]
-    }
+  if (window != null && !(window instanceof BrowserWindow)) {
```

Comment:
This is clearer than before.

Reading the code as it is now, this test looks very similar to the one in the next block five lines down, `if (window && window.constructor !== BrowserWindow)`. *If* these are equivalent, they should be folded together for readability because right now it scans as if there is a subtle, uncommented difference between the two tests.

Nothing personal in throwing this back yet again. This code is already an improvement over what's in master.

Reply:
There _is_ actually a subtle difference - the first one checks if it's anything other than a BrowserWindow, the second one if it's specifically an object 👍

Reply:
It does now but didn't when I made that comment :smiling_imp: 

Thanks for working through all these improvements and adding tests. :green_heart: